### ISSUE TITLE
EN Screenshots for you to use!

### ISSUE THREAD

**Issue by Questionguy663** [Toukibi/ToSAddon#4]
Thank you very much for the work that went into these Addons, and compatibility for EN (IToS)  

I would like to give you some English Screenshots to use as you mentioned on the Addon Manager you would like some but do not have any.

Let me know if you need any more pictures or anything else I can do for you.

# Shop Helper

![japmodshop1](https://cloud.githubusercontent.com/assets/19497435/24783558/fe6fa74c-1b44-11e7-9bf5-9ba5eaca29f6.jpg)

![japmodshop2](https://cloud.githubusercontent.com/assets/19497435/24783630/6553da82-1b45-11e7-9d27-2ce2995c3363.jpg)

![japmodshop3](https://cloud.githubusercontent.com/assets/19497435/24783707/c88023ea-1b45-11e7-90ea-6aa40dc3e354.jpg)

![japmodshop4](https://cloud.githubusercontent.com/assets/19497435/24783735/06b8090c-1b46-11e7-9e61-d181b48e42ea.jpg)

![japmodshop11](https://cloud.githubusercontent.com/assets/19497435/24806127/bc5b934a-1bab-11e7-922c-69c9caa07ce2.jpg)

![japmodshop12](https://cloud.githubusercontent.com/assets/19497435/24806130/bef3af84-1bab-11e7-84cc-60db06911fdd.jpg)

# Buff Counter

I hope one day you will consider EN version of this Addon.

![japmodbuffcounter](https://cloud.githubusercontent.com/assets/19497435/24783802/77a2ae6a-1b46-11e7-9b49-4e9b0417f748.jpg)

# Duration Notice Mini

I also hope one day you will consider EN version of this Addon too!

![japmoddurnoticemini](https://cloud.githubusercontent.com/assets/19497435/24783851/be506bc2-1b46-11e7-846a-090b586bcfe9.jpg)

# Map Mate

I'm very sorry to report that Map Mate does not work EN (IToS) /mapmate or /mmate or /mmate en has no effect.

I have tried Deleting all addons except Map Mate. It still will not work alone.

![japmodmapmap1](https://cloud.githubusercontent.com/assets/19497435/24783924/38d967ea-1b47-11e7-8d9f-0b7fde284702.jpg)

**Comment by Toukibi** [comment]
Thank you for sending lots of screenshots. I understood very well how it is displayed in IToS.
I think that the part written in Japanese probably all dots because Japanese fonts are not included. I would like to update the next time including that point.

**`BuffCounter`** was a highly appreciated add-on in Japan. Therefore, I am planning to make it multi-lingual. **`DurNoticeMini`** is also similar.

Regarding **`MapMate`**, I got contact from other people, but it seems that it is not working due to some reason. In order to solve this problem, it seems necessary for me to install IToS. It may take a while for the solution.

If I have a request from me, since I have only gone a little abroad, I do not fully understand the original nuances of English.
Therefore, I don't know how to write short sentences to fit on the screen. Also, I think that there are also misinterpretations. Therefore, I'm very happy that you can send images of those revisions. (For example, "**Within Average price range**" can be written as short as you write)

Thank you.

**Comment by Questionguy663** [comment]
If I use the Addon Developer Console, is there anything I can type in that would help trouble shoot MapMate? Another Addon maker had me do this [http://i.imgur.com/PxGqNz8.jpg](url) to help him fix something. 

I hope this will mean you don't have to necessarily install IToS if I can test for you, saving you some time and effort.  
![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24805431/aca4f452-1ba9-11e7-9975-0fb3665e7a53.PNG)


I have some Ideas for localization!  

----------

![japmodshop5](https://cloud.githubusercontent.com/assets/19497435/24803906/9fba4e4a-1ba4-11e7-90ab-d39c63e473a7.jpg)
This can be written as "Average Price Reset" or "Median Price Reset"

-------------

![japmodshop6](https://cloud.githubusercontent.com/assets/19497435/24804048/1af51acc-1ba5-11e7-82d4-14f471bb2768.jpg)
"Not Regret?" could be "Warning!" instead.

--------------

![japmodshop7](https://cloud.githubusercontent.com/assets/19497435/24804180/8b2c661a-1ba5-11e7-8eaa-a535030d980a.jpg)

"Price median unaltered" or "Price average unaltered" is as short as I can make it.  

--------------

![japmodshop8](https://cloud.githubusercontent.com/assets/19497435/24804392/2e9080ca-1ba6-11e7-85d8-aeb8bc423aeb.jpg)

"Currently ongoing" could be "Already Buffed" , "Already Brought" , "Already Running" or "Already Purchased"

-------------------

![japmodshop13](https://cloud.githubusercontent.com/assets/19497435/24806213/03f8b8ae-1bac-11e7-9198-ae0cb376e601.jpg)

I do not think White is a good colour for the text here. It is very hard to read in the game.

Instead of: 
"Cost Price 400 Rip-off (Average Pricex2.34 Cost Price)" 

Could be written as: 
"400 Silver Rip-off (Avg  x2.34 Silver)" 

This way is much shorter, Cost price can be written as Silver any time and Avg is widely recognized shorthand for average.

----------------------

![japbuffcounter4](https://cloud.githubusercontent.com/assets/19497435/24804760/73cc5dac-1ba7-11e7-85f2-1ff3fba7f5c2.png)

For Buff Counter we would call this one Top Row, a good short symbol would be "1st" or "T"

![japmodshop9](https://cloud.githubusercontent.com/assets/19497435/24804896/d78f4b92-1ba7-11e7-9b77-4265e9504c03.png)

This one we would call Second Row, short symbol would be "2nd" or "S"  

-------------

Let me know if I can help with anything else.

I sent all the pictures for you to use for the addon manager like you asked so the EN Preveiw can look like the JP Preveiw, I even have the cursor on the same thing! I am not very skilled so I am looking to help out the only way I can.

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24805738/b8732136-1baa-11e7-9989-d1a2bf5b452d.PNG)

![japmodshop10](https://cloud.githubusercontent.com/assets/19497435/24806508/021fa708-1bad-11e7-8470-640d210d8ffc.png)

**Comment by Toukibi** [comment]
Thank you for your response. And thank you for many screenshots.
Now, I am promoting MapMate's multi-language.

'Already Buffed', it was very impressive.

**Comment by Questionguy663** [comment]
Hello! Today I have updated to Buff Counter v1.1.0 and Shop Helper v0.8.1

# Shop Helper

![japmodshop14](https://cloud.githubusercontent.com/assets/19497435/24828493/439c45d4-1c56-11e7-8df3-97bd987cc1af.jpg)

I noticed that it doesn't tell you that you can type /sh to open the options in the game or on the Addon Manager, I think some people will find that helpful to know about /sh .

---------

# Buff Counter

![japmodbuffcounter2](https://cloud.githubusercontent.com/assets/19497435/24828605/7e3db2a2-1c58-11e7-9d1e-5a2ec54431e1.jpg)

The S is aligned a little lower than the T, don't worry if that cannot be fixed, it is a very minor thing!

-------------------

![japbuffcount5](https://cloud.githubusercontent.com/assets/19497435/24828611/9b684572-1c58-11e7-9d64-3f7b5c8eeafa.jpg)

Do you like this layout? I think it looks very clean and clear like this:

- Less Titles (Mode, Background, Gauge Styles)
- Consistent Alignment (Titles left, choices middle, check mark right)

This is even how Microsoft do it on Windows10
![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24828858/84e73186-1c5e-11e7-9502-6dbe2faaf82c.PNG)
![japbuffcount99](https://cloud.githubusercontent.com/assets/19497435/24828859/8723c270-1c5e-11e7-8fa9-1740a082c29c.jpg)


-----------------------

![japbuffcount6](https://cloud.githubusercontent.com/assets/19497435/24828638/362aec68-1c59-11e7-9c6f-7181292effee.jpg)

For localization how about "Lock position" most games write it like that because fix can mean repair, you could say "Fixed position" though too.

---------------

 # Duration Notice Mini & Buff Counter Bug

![japmodsboth](https://cloud.githubusercontent.com/assets/19497435/24828662/e2617f24-1c59-11e7-946d-1db5a187e385.jpg)

Did you know that after a cut scene cinematic both of these Addons become invisible until you change character or change map?

---------------------------

 # Addon Manager Description Buff Counter

![japmodmanager](https://cloud.githubusercontent.com/assets/19497435/24828694/80aa6c7c-1c5a-11e7-98c7-9806b0f36e5f.jpg)

You now have English Mode, This sentence can be deleted! 
![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24828698/a1218152-1c5a-11e7-8c0c-b87dc4827128.PNG)

-------------

 # Addon Manager Description Shop Helper

![japmodmanager2](https://cloud.githubusercontent.com/assets/19497435/24828733/a6a4d7fe-1c5b-11e7-918b-e0fdba3bbc39.jpg)

Some of these features you have implemented now so they are not upcoming changes.
![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24828737/bcf2d89e-1c5b-11e7-8810-d9c7b65b5e6d.PNG)

I'll write it all out again for you to save you some time with copy&paste ! 


# Description For English Version

### What is this?

This addon will tell you whether the price of a buff/repair/ect is expensive or cheap.
English version images coming soon.

### Features 

- Display the silver cost of players shops highlighting the reagent cost.
- Comparison of the median price of other shops you have visited.
- Coloured text on a cost gradient.
- Warning Icon for abnormally high prices shops considered a scam.
- Context menu additions allowing you to mark shops in the game world as Favorites/Blocked Players stalls.

### Installation Instructions

Install via the Addon Manager. Leave bug reports and other issues on my Github page.

### Possible Future Features

- Hide the player only whilst holding the Alt Key.
- Other player shops (Gem Roasting, Enchanter, Gender Swap, Ect)

--------------------

Let me know what you think or if I didn't explain anything very well.

**Comment by Toukibi** [comment]
Wow, thank you for your checking.

In the first:
Layout of context menu of DurNoticeMini, I feel it is very cool, too.
But there is one problem for implement this. In Japanese English characters and English it is different in character width due to the difference in font.
So, may I tell you how many space characters were used for each item?

Secondary:
I have also confirmed the problem that BuffCounter and DurNoticeMini become invisible after a cut scene cinematic. However, I have not yet found an event name to return from that state.

Thirdly:
Thank you for writing the parts of README.md. I will reflect it soon.

Finally:
In my environment (JToS), I can open the options screen without problems with the `/sh` command.
I don't know exactly what is causing it by information just now. But I will keep it in mind.

**Comment by Questionguy663** [comment]
1.  Sorry I cheated and used Microsoft Paint to move it all around! :(

![japbuffcount98meta](https://cloud.githubusercontent.com/assets/19497435/24829384/61a287de-1c68-11e7-8e59-fa1e75c1d751.jpg)

-------

2. I see, you already knew about that problem.
This addon https://github.com/axjv/Input-Switcher/tree/master/addons/cdtracker "Cooldown Tracker"
Is on the IToS part of the addon manager and the addon returns after a Cinematic cut scene, perhaps his code has some answers.

-----

3. Glad I could help, I could rewrite the other English Descriptions for the other addons if you would like me to!

-----

4. I am sorry I didn't mean that it does not work, I mean that the instructions to access the options aren't written anywhere.

![japmodshop98](https://cloud.githubusercontent.com/assets/19497435/24829111/dd8ec268-1c63-11e7-89e5-fe619f6072a0.jpg)

something like this to show people how to access the options screen.

**Comment by Toukibi** [comment]
Now, I update `DurNotice Mini` and `BuffCounter`.
Both Add-on implement English display mode and bug-fix that become invisible after a cut scene cinematic.
I finally installed IToS to adjust the width of the space, set the download while going out with shopping.
And, I checked MapMate. But in my computer, MapMate worked normally.

Therefore, I think that other factors are bad, so I will investigate in more detail.
![image](https://cloud.githubusercontent.com/assets/25131897/24838198/49f3e760-1d7e-11e7-8508-a3a27a87f14b.png)

**Comment by Toukibi** [comment]
Today, thanks holysnipz's help, I was able to fix problem that MapMate cannot start without `setting.json`.
The latest release is MapMate ver0.16.  

May I ask you to check whether it works properly in your environment?

**Comment by Questionguy663** [comment]
Hello! Today I have updated to Map Mate v0.1.6, Dur Notice Mini v1.1.0, Buff Counter v1.1.3

------------------

#Map Mate 

This Addon Instantly works today! The only small thing I can think of is the clock is exactly 1 hour wrong. (Daylight Savings Time?) I can always just remember it is 1 hour wrong if there is no way for the addon to read the time from Windows10.

---------------------------
#Dur Notice

Thank for you the English translation! and now appears after Cinematic Cut Scene!

-------------------

#Buff Counter 

Menu looks really nice now, I am surprised you did not align all the Green Ticks on the right but it looks very clear still how it is now. appears after cut scene too! 

------------

Was some code from the CoolDownTracker addon the solution to the addons disappearing after cinematic cut scenes? Until you invent a new addon I think we are done. Let me know if you would like me to look at anything else though!

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24871817/d295086c-1e12-11e7-9cce-046f481f098e.PNG)


@holysnipz I have the same problem with chatextends, https://github.com/torahamu/TOSAddon/tree/master/chatextends the addon works for some people and not for others, when it loads up a windows CMD prompt looking thing it instantly causes a crash. 

There seems to be some dependency missing on some computers that allows it to work for some and not others.

**Comment by holysnipz** [comment]
@Questionguy663 not sure about that, sorry. tbh, i'm having a CTD issue too. 
but i remove the `auto-create-directory` thing then repack for my own uses, no problem so far. (not a proper way to fix, needed to manually create directories. but it's work for me, i don't care lol)

Btw, you should create issue in torahamu's repo, see if he know about that issue or not. 
_But i don't think he speaking English, only Japanese._

**and i'm not IllFatedGaming in iToS's forum, but i do know about his thread :wink:**

**Comment by Questionguy663** [comment]
I've become a messaging service! 
![emoticon_0009](https://cloud.githubusercontent.com/assets/19497435/24877074/e687309e-1e25-11e7-97a6-22f3e8bc5b89.PNG)

for real though, how do I cut-cut-scissor-snip that auto create directory stuff, not being able to see what I am typing AND having my chat not show the end of most peoples sentences is driving me to suicide.

I will create the issue thread on his github, but also please save my life with a temp version!

https://youtu.be/cqhBq9JNX_A

"The syntax of the command is incorrect" is what the CMD prompt says 

I also need to make a issue thread on the creator of Rader for a 21:9 aspect ratio monitor issue, so much to do

EDIT: Rader is discontinued / removed, guess I am stuck with that bug forever. When you turn on "draw on minimap" (important feature) it puts it where the minimap would be on a **16:9 monitor**, not where it is on a 21:9 monitor

![emoticon_0021](https://cloud.githubusercontent.com/assets/19497435/24877325/b95bcd2c-1e26-11e7-8e21-addec4f11837.PNG)

**Comment by Toukibi** [comment]
I seemed to be working properly. I am planning to fix ShopHelper after I came home today.

What is the cause of MapMate's time flaw? Function may be slightly different between JToS and IToS.
Thanks to IToS installation I could have official source files, so I will look inside.

**Comment by Questionguy663** [comment]
Hello! I think the clock will be difficult to change unless the addon can read the computer clock

KToS 1 Time zone
JToS 1 Time zone
IToS 2,3,4,5,6+ Time Zones

-------------

The clock on their website and in the game with no addon is set for 2:21 AM EDT (North America) https://www.timeanddate.com/worldclock/usa/washington-dc

![north america](https://cloud.githubusercontent.com/assets/19497435/24944050/41ea2fb4-1f51-11e7-97fe-351dd6769002.png)

![north america2](https://cloud.githubusercontent.com/assets/19497435/24944060/4797d560-1f51-11e7-92cc-ad795bf4cecd.png)

---------------

at the same time for me it is 7:21 AM BST

![england](https://cloud.githubusercontent.com/assets/19497435/24944218/173b77ae-1f52-11e7-9b59-afd90736c1f7.png)

![england2](https://cloud.githubusercontent.com/assets/19497435/24944223/1b0d8fd4-1f52-11e7-9a75-97084bd8c143.png)

And this is just for England ( My Country) https://www.timeanddate.com/worldclock/uk/london also connecting to the Europe server is France, Germany, Russia and more all with their own time zones.

-------------

I think this is too much to ask, it is only a small detail after all. I can always look at my Telephone or Watch to get the time.

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/24944367/c43f3eea-1f52-11e7-9317-d5399c7abfbe.PNG)

**Comment by holysnipz** [comment]
Is the bottom-left corner one you're talking about? 
not sure why you got that 1hr offset problem, Its seems fine here in Telsiai.

![](http://i.imgur.com/JddRud2.png)

**Comment by Questionguy663** [comment]
After yesterdays patch(es) it's millions of hours off **Local time** :( 

Time ingame is **now** the North American (EDT) time with or without the addon. This is probably part of their fix for the dungeon instances crashing yesterday.

When I first installed and posted about the time with the addon installed it showed a clock that was 1 hour off of **English**  local time (BST-1 instead of EDT) which was really really cool.

----------------

Any chance of sharing that personal fixed chatextend that you have with the CMD prompt removed?
![emoticon_0009](https://cloud.githubusercontent.com/assets/19497435/24952642/1d746950-1f70-11e7-9341-a5dff4fd2730.PNG)
It could be months, years or never to get the official one fixed.

**Comment by holysnipz** [comment]
> Time ingame is now the North American (EDT) time with or without the addon. This is probably part of their fix for the dungeon instances crashing yesterday.

I think i see why i didn't encountered with the time offset problem. 
Because IMC never adjusted in-game clock to a local time here. 😢 

#
And sorry for the chatextends, right now i'm on my trip. But i'll leave you with a [hint](https://github.com/torahamu/TOSAddon/blob/master/chatextends/src/zchatextends.lua#L967) instead. 
Just [comment out](http://www.lua.org/pil/1.3.html) the lines that have `os.execute` command, then save as `zchatextends.lua` then replace it in the original `chatextends.ipf` by using [IPFUnpacker](https://github.com/r1emu/IPFUnpacker) and [IPFSuite](http://www.tosbase.com/downloads/IPF_Suite.zip.html)

Or just hope that IMC will push this patch into jToS soon, so he'll encounter with this problem, and find a way to fixed it. lol

**Comment by Questionguy663** [comment]
Yes, I really think IMC IToS fooling with the clock and dungeons reset time has desynced all sorts of game functions, [EU]Fediman even starts lagging horribly and suddenly when the real dungeon reset time would have been (if they hadn't changed it)

----------- 

I checked for updates to addons before starting my endeavour to download and run all these new programs no new addon updates.  
  I find out that IPFUnpacker needs some other wacky stuff "mingw64 and gcc" whatever they are, but i download a different version that apparently doesn't require them from Tehseph https://www.reddit.com/r/treeofsavior/comments/4ktlsq/need_help_with_decrypting_ipf_file/ 
  I can't figure out how to even get out of C:/windows/system 32 in the CMD prompt, have to look up a guide on the net.  
  500 mistakes later and taking hours to do something that someone else could have done in seconds I go to check my Masterpeice, my Michaelangelo's David....
  The author has just made an update to the addon and it probably now works. He was fixing it at the exact time I was fixing it across the globe.


![emoticon_0009](https://cloud.githubusercontent.com/assets/19497435/24961302/4316cc32-1f90-11e7-8627-6bff5bc54f70.PNG)


The authors comments suggest something like "now puts chat log into screenshot folder, crash was due to trying to make folder with the game running, please make the folder first." something like that. 

----------------


Toukibi I hope you are well. You haven't posted for a few days 
![emoticon_0002](https://cloud.githubusercontent.com/assets/19497435/25004974/d3d9bea0-204e-11e7-8ad7-2ae72b6995ac.png)

**Comment by Questionguy663** [comment]
I have done some more of these descriptions, if you want them.

### Buff Counter English Description localization.

# Description For English Version

### What is this?

This addon will show you the remaining number of Top Row Buff Slots left.
English version images coming soon.

### Features.

- Display the number of Top Row Buff Slots left as a Continuous Bar or Blocks 
- Display the number of Top Row Buffs Slots left numerically
- Days Remaining on your TP Token
- Other stats of buffs (I.E Blessing's Damage , Team Lodge Level, Amount of spells Divine Might Effects.)

### Installation Instructions.

Install via the Addon Manager. Leave bug reports and other issues on my Github page.

### Settings.

Settings can be accessed by Right Clicking on Buff Counter, and via these Slash Commands.

Command 	                                Effect
/buffc	                                        Toggle visibility.
/buffc reset	                                Reset the all settings.
/buffc resetpos<br>/buffc rspos	Reset the position of this add-on.
/buffc refresh	                                The Buff Counter displayed will be refreshed.
/buffc update	                                The Buff Counter displayed will be updated.
/buffc ?	                                        Display the help text in the chatlog.
/buffc jp<br>/buffc ja	                Switch to Japanese mode.
/buffc en	                                        Switch to English mode.
/buffc kr	                                        Switch to Korean mode.

---------------------------------------------

### Dur Notice Mini

# Description of English Version

### What is this?

This addon will tell you the minimum durability of your equipment, This will help you avoid leaving town only for some of your equipment to become Yellow soon after starting.  

This addon is a miniaturized version of durability value viewer 'Dur Notice' made by 'zealot

### Features

- Displays the minimum durability of your equipment
- Displays the durability as a bar with a colour gradient from 0 - 200%
- Addon can be moved to any location by dragging (when unlocked)
- Miniture size neatly fits in many areas such as under the HP & SP Bar.

### Installation Instructions.

Install via the Addon Manager. Leave bug reports and other issues on my Github page.

### Settings.

Settings can be accessed by Right Clicking on Dur Notice Mini, and via these Slash Commands.

/DurMini           
/DurMini reset          Reset the all settings
/DurMini resetpos    Reset the position of this addon
/DurMini refresh      The durability value displayed will be updated
/DurMini update      The durability value displayed will be updated
/DurMini ?                Display the help text in the chatlog.
/DurMini joke

**Comment by Toukibi** [comment]
Sorry I have not heard from you for a long time. I was updating ShopHelper and the correction of word or phrase is almost complete.

Thank you for descriptions of DurNoticeMini and BuffCounter. I will fix it in the near future.

By the way, as for the time of MapMate's clock, as a result of the investigation, it seems that the time displayed in the Map is displaying the server time. And MapMate only display the original time display small. So, we can not do the display of time as it is. As it is ...

However, if it is only the time displayed in the Map (the time of the chat log is rather troublesome and will conflict with ChatExtend), I think that it is possible to display the time of our computer.
I'd like to include it in MapMate if you like.

**Comment by Questionguy663** [comment]
That sounds really cool! 

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/25028424/3840500e-20ac-11e7-933a-e6f8d6bdbd06.PNG)


It's helpful to have the computer time to set goals like "I will play until Lunchtime" or "I should stop playing too much and stop at 6pm" it's so much harder to do this with the North American server time.

Torahamu of Chatextends seems like a nice person too, perhaps he will collaborate and use the same idea to allow the use of local time in the chat as well.

**Comment by Questionguy663** [comment]
I was awake last night thinking about these things, so I drew them today.

![japminimap2a](https://cloud.githubusercontent.com/assets/19497435/25044633/9192364a-211f-11e7-9342-a2b8f405bf46.jpg)

The different clock time could be coloured to show which mode it is in. 

------------

![japminimap3a](https://cloud.githubusercontent.com/assets/19497435/25044555/33f06c82-211f-11e7-995a-5a90fa1f99ac.jpg)

The server time as 24 hour (13:00 - 24:00 Digital) with a Blue "S" for Server time.

------------------

![japminimap3b](https://cloud.githubusercontent.com/assets/19497435/25044560/36bc0110-211f-11e7-9bcc-a5157b3f0577.jpg)

The local time as 24 hour (13:00 - 24:00 Digital) with a Gold "L" for Local time.

--------------

![japminimap4](https://cloud.githubusercontent.com/assets/19497435/25044563/3921d8e4-211f-11e7-839c-2aa32ffe00fe.jpg)

Here are a few ideas I drew  for the menu too.


![emoticon_0002](https://cloud.githubusercontent.com/assets/19497435/25044570/4599fa5c-211f-11e7-9446-4cf0f3225e32.png)

**Comment by Toukibi** [comment]
Is it classified by color? I was thinking the same thing.
Regarding the menu, I hope that it is possible to display it when right clicking on the time part, if possible.
If impossible, I will add it to the existing menu. However, in that case, the menu may be compiled into a submenu so that the menu does not become too long.

Postscript: I have not tried it with IToS yet, but I succeeded in displaying the computer's time in the mini map.

**Comment by Questionguy663** [comment]
It all sounds great to me! 

Colour means less titles, and that saves space.
Smaller menus are always better, right clicking the clock is the best way, like you said.

I am very happy it's working on your test!

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/25045096/4d1986dc-2122-11e7-8b66-a6fb6945c275.PNG)

**Comment by Toukibi** [comment]
Today I released DurNoticeMini and MapMate. MapMate was able to finish as expected.
Please try by all means.

While display local time, the clock become light blue.

**Comment by Questionguy663** [comment]
I have updated DurNoticeMini and MapMate.

### MapMate  
The clock is perfect, well done! Thank you! It's so cool to be able to have local time. I tested all modes before 13:00 and after 13:00. 24 Hour mode works well and so does Server time and Local Time.

The only thing I can think of is that we usually write AM and PM after the time, You don't need to worry about this small thing though.

![japtime1](https://cloud.githubusercontent.com/assets/19497435/25063449/891dea78-21dc-11e7-97c2-c23faec54174.jpg)

Google.

![japtime2](https://cloud.githubusercontent.com/assets/19497435/25063451/8cab8182-21dc-11e7-8363-773548fca11e.jpg)

Windows.


### DurNoticeMini
I am not sure what things changed with todays patch, what was in the change log?

![emoticon_0002](https://cloud.githubusercontent.com/assets/19497435/25063419/d5cb60a4-21db-11e7-8df4-a19926716296.png)

**Comment by Toukibi** [comment]
The changes of DurNoticeMini are really small things. The following problem has been fixed in this patch.

When we use DurNoticeMini with default location, and after cinema effect, the location of DurNoticeMini moves down 30px.

**Comment by Questionguy663** [comment]
I've been playing the game using the addons, I think the clock is my **Favorite** addon now as I no longer queue for a dungeon right when food is about to be cooked.
![emoticon_0006](https://cloud.githubusercontent.com/assets/19497435/25088102/a223bbde-236b-11e7-999d-e43c52eb9045.png)

------------

Some small bugs, probably unavoidable:

### Shophelper

Only says that you have a buff when you buy the next one, or reopen the shop. I think this is unavoidable as it would need a refresh rate putting extra strain on resources.

### MapMate

When changing maps clock goes from Blue to White for a while, then refreshes to Blue again. Probably unavoidable as having extra refreshes would use up system resources again.

---------------

I think everything is in a really good place already! I am looking foward to your next Addon.

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/25088164/1a648146-236c-11e7-97ca-5653b49141da.PNG)

**Comment by Toukibi** [comment]
Thank you for your reporting.
The bugs of MapMate is easy to fix, and now I fixed it on my computer. I will release it on the next update.
But the bugs of ShopHelper is unavoidable, because of my skill shortage, sorry.

Well, I am currently extending MapMate's functionality significantly. I am sorry for the screenshot in Japanese version, but I add the following things.
* A mark appears at the position of the treasure box with Map and MiniMap, and you can know whether that treasure box is opened or not
* You can see the list of NPCs you have not met yet
* You can see a list of the enemies that live in that map, and also you can know the max population count and the respawn time

Since it takes a lot of time to finish it cleanly, I am thinking to make further updates after releasing it with insufficient appearance.
![treasurebox](https://cloud.githubusercontent.com/assets/25131897/25088967/7dc3c672-23b5-11e7-89a9-57590bb28d3b.jpg)

**Comment by Questionguy663** [comment]
Your new features are amazing! I'm looking foward to having them!
![emoticon_0001](https://cloud.githubusercontent.com/assets/19497435/25089933/7495cf88-2378-11e7-8a66-a6ba2e8773fd.png)

------

My Idea was not as good.

![japsystemmate](https://cloud.githubusercontent.com/assets/19497435/25090039/12071592-2379-11e7-8faa-80ddc707c799.jpg)

Just a hook to make FPS and Ping Dragable.

-----

### DurNoticeMini

In this screenshot Dur Notice Mini says my most broken armour is 3 (Necklace) but my my most broken armour piece is 0 (Boots) Is this the correct behaviour? does it not count broken (0) pieces?

![20170417142506_1](https://cloud.githubusercontent.com/assets/19497435/25090243/0a222924-237a-11e7-8c49-04a8e3c40653.jpg)

**Comment by Toukibi** [comment]
Ha ha ha, temperature is not possible indeed.

About DurNoticeMini, I try not to display the durability of equipment that has broken as you say.
However, there are circumstances that I have decided on having to worry about such specifications. It is also a matter that we cannot know that it has broken.

So I will fix it so that it can be switched with the right click menu.

**Comment by Questionguy663** [comment]
No need to change it, I was just unsure if it was part of the design. Most of the time I repair broken items and then look, I just happened to look before repair!

-------------

I was thinking, Japanese Tree of Savior has the Hunting Grounds patch now right?

With the hunting grounds patch, and the Addon that makes the "Fog of War" into Red squares and the Addon Minimap Mate we have the Zone Map name written three times in total.

![japmapmate1](https://cloud.githubusercontent.com/assets/19497435/25132429/76fe09c4-2441-11e7-8891-8fc35f261edd.jpg)


Even with uninstalling the other addon it will still be written twice. Perhaps the space can be used for something else now. What do you think of this idea?

**Comment by Questionguy663** [comment]
Hello! I was waiting to see if you replied for a few days. I have updated to Map Mate v0.7.0 **and then to** Map Mate v0.7.1 


I have an important first question before I discuss more, did you hide the ToS Official Map name below the map?

![japminimap5](https://cloud.githubusercontent.com/assets/19497435/25257925/82c44c02-2633-11e7-90fe-dec7f0b470ca.jpg)

**Comment by Toukibi** [comment]
Hi. I was exhausted from updating MapMate and could not respond.
The new MapMate ver 0.71 implemented the following features.
* Implement Fog on the map and mini map. So we don't have to use MapFogViewer and so on.
* A mark appears at the position of the treasure box with Map and MiniMap, and we can know whether that treasure box is opened or not
* We can see the list of NPCs we have not met yet. And addon displays them thinly on the map.
* We can see a list of the enemies that live in that map, and also we can know the max population count and the respawn time.
* In a place where mini-map can be displayed, hide map name display by official patch.

So, the answer is Yes. I hide the ToS Official Map name below the map while the minimap is visible.

**Comment by Questionguy663** [comment]
Thank you for all the hard work! 
![emoticon_0001](https://cloud.githubusercontent.com/assets/19497435/25305755/fced9e02-2778-11e7-9662-0525acf1add4.png)

I just want to talk about the map names for a little longer, then i'll talk about all the other features!

In the English version of tree of saviour we have some very very long map names, he are some example screenshots of one of the longest ones.

![japminimap6](https://cloud.githubusercontent.com/assets/19497435/25305781/7f747e9a-2779-11e7-9376-6a520265e383.jpg)

I'm sure this is much shorter in Kanji (did I say this right?) 	ルクソナ再建団東館

http://toscamp.com/map/mapviewer.php?map=d_firetower_61_2&lang=jp Japanese
http://toscamp.com/map/mapviewer.php?map=d_firetower_61_2&lang=en English
-----

When you look at how this is displayed in game with MapFogViewer Ex you see this

![japminimap7](https://cloud.githubusercontent.com/assets/19497435/25305826/637b6bc6-277a-11e7-9b1f-93dac73c2d18.jpg)

The full name is displayed and uses an extra line, quite good, but not perfect as it is behind another asset.

-------

When you look at it on the official Map Name it looks like this

![japminimap8](https://cloud.githubusercontent.com/assets/19497435/25305848/b98e7c9c-277a-11e7-9ab5-a34adba256af.jpg)

As you can see the map name is actually cut short! oh no!

![emoticon_0016](https://cloud.githubusercontent.com/assets/19497435/25306869/76061b74-278e-11e7-8b9a-36b4f8eb338c.PNG)


----------------

When you look at the map through Map Mate it displays like this:

![japminimap9](https://cloud.githubusercontent.com/assets/19497435/25305867/1839f366-277b-11e7-91c7-c21c125c9104.jpg)

The full name is there which is good, but there is some collision with the Clock and Map Zoom Out buttons.

------------

And there is one other intresting feature of the Official Map Name. It also shows the Area Names! Here is a small segment of Roxona Reconstruction Agency East Building:

![japminimap10](https://cloud.githubusercontent.com/assets/19497435/25305904/bed54676-277b-11e7-8982-17ea6747849b.jpg)

![japminimap11](https://cloud.githubusercontent.com/assets/19497435/25305916/f06d86ee-277b-11e7-82b8-46b1c2be289d.jpg)

As you can see the Official Map Name has added the sub area Reception Room.

![japminimap12](https://cloud.githubusercontent.com/assets/19497435/25305933/3da3ee1c-277c-11e7-8592-feaef9ab84ca.jpg)

And now that I have moved to the Northern Transportation Device the Official Map Name has written this sub area on my screen.

---------------------

Here I will draw what I think could be the best using all designs!

![japminimap13fff](https://cloud.githubusercontent.com/assets/19497435/25306660/9de907d2-2789-11e7-8411-6ee134889a8e.jpg)

1.  Moved the Star difficulty down to the next line, the nice thing about the stars line is that it doesn't change size on 1 Star, 2 Star, 3 Star or 4 Star.

2. Kept the Dungeon Icon / Town Icon. This can be blank when it's a normal map.

3. Name below the map, there is more space here for the long names so they don't get cut off. I like your small font that you use the best.

4. Sub Area Name I think this is cool, but if this can't be implemented it's the least important one.

**Comment by Toukibi** [comment]
In Japanese version (JToS), "Roxona Reconstruction Agency East Building" should look like the following screenshot.  
![image](https://cloud.githubusercontent.com/assets/25131897/25332742/95c2ae7e-2922-11e7-94cd-cc947293bf67.png)  
It is fairly short. So it didn't come to mind that it would be too long and overlap.  
Therefore, MapMate also changed it so that it wraps back when Map name is too long.  
![image](https://cloud.githubusercontent.com/assets/25131897/25333224/696d22e4-2924-11e7-8f13-f08e3f190fb7.png)  
In this screenshot Map name has been written five times on purpose.
***
Although it is display of the area name, it was easy to display the display.  
However, it is not easy to finish the appearance satisfactory.  
![image](https://cloud.githubusercontent.com/assets/25131897/25333577/bbe26d58-2925-11e7-86ec-217098609285.png)
![image](https://cloud.githubusercontent.com/assets/25131897/25333756/8db5fcd2-2926-11e7-875b-b38b4d637c1a.png)
![image](https://cloud.githubusercontent.com/assets/25131897/25333692/3e1cbb5c-2926-11e7-8ac4-51c6c03952b3.png)
![image](https://cloud.githubusercontent.com/assets/25131897/25333790/c34e44b2-2926-11e7-8b53-54e565bd0d28.png)
So, I think that trying to implement a option to select whether to display inside MiniMap or outside.

**Comment by Questionguy663** [comment]
I think you will make the best decisions about the final appearance, I just wanted to make sure that you had all possible information before making the final choice.

![emoticon_0006](https://cloud.githubusercontent.com/assets/19497435/25343465/09eb51b4-2907-11e7-9bd9-7253d5dc553f.png)

If you can have manually overwritten names with a settings file then when A player steps onto "Roxona Reconstuction Agency East Building" it could display as **"Roxona Rec. Agency E. Building"**

If that was possible I could submit a list of long names that could have short names in the settings file and they would fit!

-----------------

Now I feel I can talk about all the other new features!

### So cool! All of them!

![emoticon_0001](https://cloud.githubusercontent.com/assets/19497435/25343706/b6615a92-2907-11e7-8566-211bc43af1da.png)

- Red 'Fog of War' Squares: means I can remove the other addon. One less map name!
- Monster List: It's really nice to have all the monsters on the map and their kill counts and respawn times in one place to mouse over. This really helps one of my friends who has trouble remembering everything!
- Quests remaining: Very nice! its so much better to see it here as a small icon than open the quest menu and count how many quests are left manually. And very good when the game surprises you with more quests when you think you have finished. ( Example Demon Prison)
- NPC's Remaining: Very cool! not only does the list and the names help out a lot, especially with the hidden ones you can also see them ghosted out on the map, this extra detail is a feature just by itself too!

![emoticon_0002](https://cloud.githubusercontent.com/assets/19497435/25343963/88001520-2908-11e7-9ee5-c9cceebfb42f.png)


---------------------

# Feature Ideas

### Track "Statue of Godess Zemyna" (+1 Stat Point Statue)

![japminimap14](https://cloud.githubusercontent.com/assets/19497435/25344151/24f78354-2909-11e7-93f7-4d8b2fd1f0bc.jpg)

![japminimap15](https://cloud.githubusercontent.com/assets/19497435/25344222/5c301ec6-2909-11e7-9692-eb37297e3809.jpg)

If she doesnt give any NPC score like these then maybe she gets her own icon? You could use this picture:

![icon_cler_cravezemyna](https://cloud.githubusercontent.com/assets/19497435/25344317/af48d788-2909-11e7-924c-1dcecabc2bb1.png)

this is litterally the spell "Statue of Godess Zemyna" from the Dievdirbys class

http://www.tosbase.com/database/skills/40402/

![japminimap16](https://cloud.githubusercontent.com/assets/19497435/25344929/81862970-290b-11e7-94d0-5e8071304790.jpg)

I drew her in with Adobe Photoshop, what do you think?

------------

### Ban Undesirable Treasure

![japminimap17](https://cloud.githubusercontent.com/assets/19497435/25345127/15c86666-290c-11e7-82aa-09a73c3d5812.jpg)

In the Crystal Mines 1F there is a level 2 Treasure chest with a "Bone Necklace" in it, it is not worth a level 2 key, how about a way to Ban it like this feature from Shop Helper.

![japminimap18](https://cloud.githubusercontent.com/assets/19497435/25345364/cf6cb1c6-290c-11e7-8ea7-3b725685585b.jpg)

It is not an important feature though, I can always just not loot the treasure!

![emoticon_0013](https://cloud.githubusercontent.com/assets/19497435/25345427/f38b66ec-290c-11e7-93ff-69047aa395ce.PNG)

---------------

### Ignore NPC's

Some of the NPC's weren't coded too well by IMC, it would be nice to put them on ignore until they are fixed.

![japminimap19](https://cloud.githubusercontent.com/assets/19497435/25346073/82a8c1c0-290e-11e7-85d8-0bc2d99c5b92.jpg)

This is NPC Sandra

![japminimap20](https://cloud.githubusercontent.com/assets/19497435/25346143/bc448414-290e-11e7-856f-ba77268090c2.jpg)

She is the only NPC I am missing, but she is bugged / broken and no matter how much I talk to her she does not complete the list.

![japminimap21](https://cloud.githubusercontent.com/assets/19497435/25346423/6d5267e4-290f-11e7-9c45-d026efcc024f.jpg)

This is my drawing of a ban list that I made up, something that we can edit if they fix them. What do you think?

-----

# Bug Reports

### Bug Monster Counts in Party

![japminimap22](https://cloud.githubusercontent.com/assets/19497435/25346536/c11920b6-290f-11e7-8dd3-8762490198d9.jpg)

When you mouse over this and can see your kill counts for the Adventure Journal sometimes it's been very wrong.
For example when I was playing with one friend each of us on a new character this number sometimes became wildly incorrect saying things like 50/300 killed, when you press F4 to look in the journal though it actually said 292/300 killed for both of us.

I'll try to get some better screenshots of this later on.

-----

**Comment by Toukibi** [comment]
Thank you for liking the new features.
I would like to answer the stated needs.
## Track "Statue of Godess Zemyna"
Yes, I also wanted to display whether or not we gave prayer to the geminar statue and I tried it already. However, the status of whether you prayed on the statue was different from the status of getting talked to another NPC.  
And finally, I could not get it now. I would like to implement it as soon as I can acquire it.
## "Ban Undesirable Treasure" and "Ignore NPC's"
It is certainly possible to change the display if it matches each ID. However, there is one fatal and serious problem in the add-on specification.  
### **Reading and writing of the setting file is very slow.**  
So, we have to minimize the setting items used for add-ons. Otherwise, its longer time will directly lead to extension of loading time, in all Map and channel changes.  
Therefore, it is difficult to realize these two demands. However, as for the NPC which can not be talked in the state of the quest, I am going to change it to exclude if it can be acquired.
## [Bug Reports] Bug Monster Counts in Party
Yes, that was a problem due to my technical shortage.  
At the time of publication, it was impossible to acquire an event at the moment of defeating a monster. Therefore, the displayed count will be the number when MapMate was finally updated. (In most cases it is the time when Map loading is complete)  
However, now it succeeded in acquiring the event, so it has become possible to display meticulous counts that have always been killed. It is already installed in the next release.
